src/Lecture4.hs

```diff
@@ -226,7 +309,24 @@ implement the next task.
 -}
 
 combineRows :: NonEmpty Row -> Stats
-combineRows = error "TODO"
+-- combineRows = sconcat . NE.map rowToStats
+combineRows = foldl' rowStatsConcat defStats
+  where
+    defStats =
+      Stats
+        { statsTotalPositions = Sum 0,
+          statsTotalSum = Sum 0,
+          statsAbsoluteMax = Max 0,
+          statsAbsoluteMin = Min 0,
+          statsSellMax = Nothing,
+          statsSellMin = Nothing,
+          statsBuyMax = Nothing,
+          statsBuyMin = Nothing,
+          statsLongest = MaxLen ""
+        }
```

Comment:
Instead of implementing a default stats value, you can pattern match on `NonEmpty` and use the first value in the list as the default. You already know that the list is not empty, so you don't need to assume that it can be empty.

Reply:
I thought about it but I didn't look for the NonEmpty constructor. Now I know about ```:|``` !